features/apt_messages.feature

```diff
@@ -669,6 +669,352 @@ Feature: APT Messages
+            },
+            "lines": [
+              "one",
+              "two"
+            ]
+          }
+        ]
+      }
+      """
+    When I run `pro refresh messages` with sudo
+    When I apt upgrade
+    Then I will see the following on stdout:
+      """
+      Reading package lists...
+      Building dependency tree...
+      Reading state information...
+      Calculating upgrade...
+      0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
+      """
+    When I create the file `/var/www/html/aptnews.json` on the `apt-news-server` machine with the following:
+      """
+      {
+        "messages": [
+          {
+            "begin": "$behave_var{today}",
+            "selectors": {
+              "architectures": ["arm64", "amd64"]
+            },
+            "lines": [
+              "one",
+              "two"
+            ]
+          }
+        ]
+      }
+      """
+    When I run `pro refresh messages` with sudo
+    When I apt upgrade
+    Then I will see the following on stdout:
+      """
+      Reading package lists...
+      Building dependency tree...
+      Reading state information...
+      Calculating upgrade...
+      #
+      # one
+      # two
+      #
+      0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
+      """
+    # Testing packages selector when package is not installed
+    When I create the file `/var/www/html/aptnews.json` on the `apt-news-server` machine with the following:
+      """
+      {
+        "messages": [
+          {
+            "begin": "$behave_var{today}",
+            "selectors": {
+              "codenames": ["<release>"],
+              "architectures": ["amd64"],
+              "packages": [["libcurl4", "==", "<installed_version>"]]
+            },
+            "lines": [
+              "one",
+              "two"
+            ]
+          }
+        ]
+      }
+      """
+    And I run `pro refresh messages` with sudo
+    And I apt upgrade
+    Then I will see the following on stdout:
+      """
+      Reading package lists...
+      Building dependency tree...
+      Reading state information...
+      Calculating upgrade...
+      0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
+      """
+    # Testing package selector when package installed
+    When I create the file `/var/www/html/aptnews.json` on the `apt-news-server` machine with the following:
+      """
+      {
+        "messages": [
+          {
+            "begin": "$behave_var{today}",
+            "selectors": {
+              "packages": [["<package>", "==", "<installed_version>"]]
+            },
+            "lines": [
+              "one",
+              "two"
+            ]
+          }
+        ]
+      }
+      """
+    And I apt install `<package>=<installed_version>`
+    And I run `apt-mark hold <package>` with sudo
+    And I run `pro refresh messages` with sudo
+    And I apt upgrade
+    Then stdout matches regexp:
```

Comment:
Many of these `stdout matches regexp` steps can be replaced by the stricter `stdout contains substring:` step